### PR TITLE
[FIX] *: make some user errors translatable

### PR DESCRIPTION
**Pull request by niyasraphy**
[FIX] *: make some user errors translatable

*: base, account_payment, l10n_ar, mrp_subcontracting, point_of_sale, pos_restaurant, purchase, sale, sale_loyality, website_sale_loyality

enterprise: https://github.com/odoo/enterprise/pull/34432

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by Julien00859 — CHANGES_REQUESTED** [review]
Various bad usages of `%` inside of translated fields as reported by the CI.

The following is a bad usage as you ask the `_` function to translate `"Hello world"` instead of `"Hello %s"`. Meaning you have to provide a translation for every possible `x`. In case `x` is user-defined, it is impossible.
```py
x = 'world'
_("Hello %s" % x)
```

The correct usage of `_` is as follow:
```py
x = 'world'
_("Hello %s", x)
```

The commit message title should always include a module name, here as you modify a lot of modules at the same time, you can change the title to `[FIX] *: make some user errors translatable`

It looks like the pot file for the `l10n_in_edit` module was missing. That should be part of another commit `[FIX] l10n_in_edit: missing pot file` as it is another kind of error. It can be inside this PR (so 2 commits here).

**Comment by niyasraphy** [comment]
hi @Julien00859 ,
thanks for the review, updated the pr with the given suggestions.

**Comment by niyasraphy** [comment]
@Julien00859 should i add the l10n_edi module to odoo/.tx/config file ?

**Comment by mart-e** [comment]
@niyasraphy no need to add .pot for l10n_ modules if they are not translated/published on Transifex. We do not want to publish translations for the Spanish chart of account and have French people to waste time translating it.

Also regarding the missing `_(...)` calls, we should keep it for error messages displayed to the enduser. For instance `"Invalid characters in table name %r"` is probably never shown to the user but only to help the developer. If you have a issue while generating db tables, chances are that you are no in a state where you will be able to retrieve translations anyway.

File: addons/mrp_subcontracting/models/stock_picking.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def _action_done(self):
             if not production:
                 continue
             if len(production) > 1:
-                raise UserError("It shouldn't happen to have multiple production to record for the same subcontracted move")
+                raise UserError(_("It shouldn't happen to have multiple production to record for the same subcontracted move"))
```
[the remark this answers is not in the document]
Comment (niyasraphy) [reply addons/mrp_subcontracting/models/stock_picking.py]: @xmo-odoo updated the message to: Multiple production should not be recorded for the same subcontracted move

**Comment by niyasraphy** [comment]
> @niyasraphy no need to add .pot for l10n_ modules if they are not translated/published on Transifex. We do not want to publish translations for the Spanish chart of account and have French people to waste time translating it.
> 
> Also regarding the missing `_(...)` calls, we should keep it for error messages displayed to the enduser. For instance `"Invalid characters in table name %r"` is probably never shown to the user but only to help the developer. If you have a issue while generating db tables, chances are that you are no in a state where you will be able to retrieve translations anyway.

hi @mart-e ,
updated the pr by removing the .pot files added for l10n_ modules, if this is the cases, l10n_ar_{} and l10n_fi modules is fine to have pot files ?

thanks

File: addons/mrp_subcontracting/models/stock_picking.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ def _action_done(self):
             if not production:
                 continue
             if len(production) > 1:
-                raise UserError("It shouldn't happen to have multiple production to record for the same subcontracted move")
+                raise UserError(_("Multiple production should not be recorded for the same subcontracted move."))
```
Comment (ticodoo) [inline addons/mrp_subcontracting/models/stock_picking.py:60]:
```suggestion
                raise UserError(_("There shouldn't be multiple productions to record for the same subcontracted move."))
```

I think there might also be a copy paste error in your corresponding pot file change too.

File: addons/sale_loyalty/tests/common.py [hunk 3]
```diff
@@ -192,7 +193,7 @@ def _apply_promo_code(self, order, code, no_reward_fail=True):
             raise ValidationError(status['error'])
         if not status and no_reward_fail:
             # Can happen if global discount got filtered out in `_get_claimable_rewards`
-            raise ValidationError('No reward to claim with this coupon')
+            raise ValidationError(_('No reward to claim with this coupon'))
         coupons = self.env['loyalty.card']
```
Comment (Feyensv) [inline addons/sale_loyalty/tests/common.py:197]: I don't think it's useful to translate error messages from tests ;)

File: addons/website_sale_loyalty/tests/test_shop_sale_coupon.py [hunk 4]
```diff
@@ -207,7 +207,7 @@ def _apply_promo_code(self, order, code, no_reward_fail=True):
             raise ValidationError(status['error'])
         if not status and no_reward_fail:
             # Can happen if global discount got filtered out in `_get_claimable_rewards`
-            raise ValidationError('No reward to claim with this coupon')
+            raise ValidationError(_('No reward to claim with this coupon'))
```
Comment (Feyensv) [inline addons/website_sale_loyalty/tests/test_shop_sale_coupon.py:210]: Same comment

File: odoo/addons/base/models/res_partner.py [hunk 5]
```diff
@@ -487,7 +487,7 @@ def onchange_company_type(self):
     @api.constrains('barcode')
     def _check_barcode_unicity(self):
         if self.barcode and self.env['res.partner'].search_count([('barcode', '=', self.barcode)]) > 1:
-            raise ValidationError('An other user already has this barcode')
+            raise ValidationError(_('An other user already has this barcode'))
```
Comment (ticodoo) [inline odoo/addons/base/models/res_partner.py:490]: If we're correcting the English for other messages then we may as well fix this one too :shrug: 

```suggestion
            raise ValidationError(_('Another user already has this barcode'))
```

File: odoo/addons/base/models/ir_attachment.py [hunk 6]
```diff
@@ -110,7 +110,7 @@ def _get_path(self, bin_data, sha):
             os.makedirs(dirname)
         # prevent sha-1 collision
         if os.path.isfile(full_path) and not self._same_content(bin_data, full_path):
-            raise UserError("The attachment is colliding with an existing file.")
+            raise UserError(_("The attachment is colliding with an existing file."))
```
Comment (ticodoo) [inline odoo/addons/base/models/ir_attachment.py:113]: Another potential English improvement

```suggestion
            raise UserError(_("The attachment collides with an existing file."))
```

File: addons/sale_loyalty/tests/common.py (hunk 3 above)
Comment (niyasraphy) [reply addons/sale_loyalty/tests/common.py]: thanks and updated

File: addons/website_sale_loyalty/tests/test_shop_sale_coupon.py (hunk 4 above)
Comment (niyasraphy) [reply addons/website_sale_loyalty/tests/test_shop_sale_coupon.py]: thanks and updated

**Review by Feyensv — APPROVED** [review]

File: addons/mrp_subcontracting/models/stock_picking.py (hunk 2 above)
Comment (niyasraphy) [reply addons/mrp_subcontracting/models/stock_picking.py]: @ticodoo thanks and updated

File: odoo/addons/base/models/res_partner.py (hunk 5 above)
Comment (niyasraphy) [reply odoo/addons/base/models/res_partner.py]: @ticodoo thanks and updated

File: odoo/addons/base/models/ir_attachment.py (hunk 6 above)
Comment (niyasraphy) [reply odoo/addons/base/models/ir_attachment.py]: @ticodoo thanks and updated

**Review by ticodoo — APPROVED** [review]

File: addons/purchase/report/purchase_report.py [hunk 7]
```diff
@@ -7,7 +7,7 @@
 
 import re
 
-from odoo import api, fields, models, tools
+from odoo import api, fields, models, tools, _
```
Comment (MiquelRForgeFlow) [inline addons/purchase/report/purchase_report.py:10]:
```suggestion
from odoo import api, fields, models, _
```
Selection_675

Comment (niyasraphy) [reply addons/purchase/report/purchase_report.py]: thanks @MiquelRForgeFlow , updated

File: addons/purchase/report/purchase_report.py [hunk 8]
```diff
@@ -7,7 +7,7 @@
 
 import re
 
-from odoo import api, fields, models, tools
+from odoo import api, fields, models,  _
```
Comment (MiquelRForgeFlow) [inline addons/purchase/report/purchase_report.py:10]:
```suggestion
from odoo import api, fields, models, _
```

Comment (niyasraphy) [reply addons/purchase/report/purchase_report.py]: updated

**Review by vin-odoo — APPROVED** [review]

**Review by trj-odoo — APPROVED** [review]

**Review by Julien00859 — APPROVED** [review]

**Comment by Julien00859** [comment]
Thanks :)
@robodoo r+

**Comment by niyasraphy** [comment]
hi @odoo/rd-ecommerce-documents if you have time, can you review this.
thanks

**Pull request closed without merging**